tempodb/encoding/iterator_wal.go

```diff
@@ -0,0 +1,44 @@
+package encoding
+
+import (
+	"context"
+	"io"
+
+	"github.com/grafana/tempo/tempodb/encoding/common"
+)
+
+type walIterator struct {
```

Comment:
Could/should this be in the wal package?

Reply:
Renamed to "recordless" iterator. Also reviewing the iterators I think "paged" and "record" iterator could be combined fairly easily.